tools/metrics.py

```diff
@@ -65,7 +66,12 @@ def __init__(self, name, dir, output, make_flags=None):
     "s": PortData("stm32", "stm32", "build-PYBV10/firmware.elf", "BOARD=PYBV10"),
     "c": PortData("cc3200", "cc3200", "build/WIPY/release/application.axf", "BTARGET=application"),
     "8": PortData("esp8266", "esp8266", "build-ESP8266_GENERIC/firmware.elf"),
-    "3": PortData("esp32", "esp32", "build-ESP32_GENERIC/micropython.elf"),
+    "3": PortData(
+        "esp32",
+        "esp32",
+        "build-ESP32_GENERIC/micropython.elf",
+        pre_make_cmd=[".", "esp-idf/export.sh"],
```

Comment:
It feels like this should be configurable by a command-line option, or environment variable.  (Eg for myself I have a shell alias `idf` which gets me the correct IDF environment.)

Reply:
In my WIP I've implemented it so that an environment variable such as `PRE_CMD_b=fnord` will supply a different PRE_CMD for the build tagged with 'b'.

Reply:
now pushed.